kong/plugins/datadog/handler.lua

```diff
@@ -0,0 +1,77 @@
+local BasePlugin = require "kong.plugins.base_plugin"
+local basic_serializer = require "kong.plugins.log-serializers.basic"
+local statsd_logger = require "kong.plugins.datadog.statsd_logger"
+
+local ngx_log = ngx.log
+local ngx_timer_at = ngx.timer.at
+local string_gsub = string.gsub
+
+local function request_counter(api_name, logger)
+  local stat = api_name..".request.count"
+  logger:counter(stat, 1, 1)
+end
+
+local function status_counter(api_name, message, logger)
+  local stat = api_name..".request.status."..message.response.status
+  logger:counter(stat, 1, 1)
+end
+
+local function request_size_gauge(api_name, message, logger)
+  local stat = api_name..".request.size"
+  logger:gauge(stat, message.request.size, 1)
+end
+
+local function latency_gauge(api_name, message, logger)
+  local stat = api_name..".latency"
+  logger:gauge(stat, message.latencies.request, 1)
+end
+
+local function log(premature, conf, message)
+  if premature then return end
+  
+  local logger, err = statsd_logger:new(conf)
```

Comment:
You `new` method does not return an error. Either return `nil, err` when `setpeername` failed in `new`, either test for 

```
if not logger then
  -- error + return
end
```

I think the 1st solution makes more sense:

```
local _, err = sock:setpeername(conf.host, conf.port)
if err then
  return nil, failed to connect to "..conf.host..":"..tostring(conf.port)..": "..err
end

-- and in handler:
local logger, err = statsd_logger:new(conf)
if err then
  ngx_log(...)
  return
end
```